src/components/Game.js

```diff
@@ -1,37 +1,200 @@
 import '../css/Game.css';
 import GameBoard from './GameBoard'
-import React, { Component, useEffect } from 'react'
-import { useParams } from 'react-router-dom'
-import GameMap from './GameMap';
-import { fetchCountryData } from '../apiCalls'
+import React, { Component } from 'react'
+import GameMap from './GameMap'
+import { fetchCountryData } from '../apiCalls';
+import { winningBoards, getRandomIndex } from '../utils'
 
 class Game extends Component {
-    constructor(props) {
-      super(props)
-      this.state = {
-        countries: []
+  constructor(props) {
+    super(props)
+    this.state = {
+      allCountries: [],
+      currentRegion: this.props.region,
+      currentRegionCountries: [],
+      currentLat: 'N/S',
+      currentLong: 'E/W',
+      nextBtnTxt: 'Get Coordinates',
+      bingoBtnTxt: 'Set Game Board!',
+      calledCoordinateIds: [],
+      currentBoard: [],
+      winStatus: ''
+    }
+  }
+
+  getCoordinates = () => {
+    let randomIndex = getRandomIndex(this.state.currentRegionCountries)
+    let randomCountry = this.state.currentRegionCountries[randomIndex]
+    while (this.state.calledCoordinateIds.includes(randomCountry.id)) {
+      randomIndex = getRandomIndex(this.state.currentRegionCountries)
+      randomCountry = this.state.currentRegionCountries[randomIndex]
+    }
+
+    let latNum = randomCountry.latitude
+    let latDir = 'N'
+    let longNum = randomCountry.longitude
+    let longDir = 'E'
+    
+    if (latNum < 0) {
+      latNum = latNum * -1
+      latDir = 'S'
+    }
+    if (longNum < 0) {
+      longNum = longNum * -1
+      longDir = 'W'
+    }
+
+    console.log('called country: ', randomCountry.name)
```

Comment:
Oh man, what a useful bit of functionality for testing. Thanks for including this.